src/SWP/Bundle/TemplatesSystemBundle/Service/RendererService.php

```diff
@@ -0,0 +1,132 @@
+<?php
+
+/*
+ * This file is part of the Superdesk Web Publisher Template Engine Bundle.
+ *
+ * Copyright 2015 Sourcefabric z.u. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2015 Sourcefabric z.ú
+ * @license http://www.superdesk.org/license
+ */
+
+namespace SWP\Bundle\TemplatesSystemBundle\Service;
+
+use SWP\Bundle\TemplatesSystemBundle\Container\ContainerRenderer;
+use SWP\Bundle\TemplatesSystemBundle\Factory\ContainerRendererFactory;
+use SWP\Bundle\TemplatesSystemBundle\Provider\ContainerProviderInterface;
+use SWP\Bundle\TemplatesSystemBundle\Widget\TemplatingWidgetHandler;
+use Symfony\Component\DependencyInjection\ContainerInterface as ServiceContainerInterface;
+
+/**
+ * Class RendererService.
+ */
+class RendererService implements RendererServiceInterface
+{
+    /**
+     * @var ServiceContainerInterface
+     */
+    protected $serviceContainer;
+
+    /**
+     * @var string
+     */
+    protected $cacheDir;
+
+    /**
+     * @var bool
+     */
+    protected $debug;
+
+    /**
+     * @var ContainerServiceInterface
+     */
+    protected $templateContainerService;
+
+    /**
+     * @var ContainerProviderInterface
+     */
+    protected $containerProvider;
+
+    /**
+     * @var ContainerRendererFactory
+     */
+    protected $containerRendererFactory;
+
+    /**
+     * RendererService constructor.
+     *
+     * @param ServiceContainerInterface  $serviceContainer
+     * @param string                     $cacheDir
+     * @param bool                       $debug
+     * @param ContainerServiceInterface  $templateContainerService
+     * @param ContainerProviderInterface $containerProvider
+     * @param ContainerRendererFactory   $containerRendererFactory
+     */
+    public function __construct(
+        ServiceContainerInterface $serviceContainer,
+        string $cacheDir,
+        bool $debug,
+        ContainerServiceInterface $templateContainerService,
+        ContainerProviderInterface $containerProvider,
+        ContainerRendererFactory $containerRendererFactory
+    ) {
+        $this->cacheDir = $cacheDir;
+        $this->debug = $debug;
+        $this->serviceContainer = $serviceContainer;
+        $this->templateContainerService = $templateContainerService;
+        $this->containerProvider = $containerProvider;
+        $this->containerRendererFactory = $containerRendererFactory;
+    }
+
+    /**
+     * @param string $name
+     * @param array  $parameters
+     * @param bool   $createIfNotExists
+     *
+     * @return ContainerRenderer
+     *
+     * @throws \Exception
+     */
+    public function getContainerRenderer($name, array $parameters = [], $createIfNotExists = true)
```

Comment:
I would split this method a bit because it has to many responsibilities and this class injects too much dependencies, maybe new class container renderer manager can be created and some code can be moved therE?